### ISSUE TITLE
Trying to get Hotone Jogg interface to work

### ISSUE THREAD

**Issue by mikeoliphant** [mdias/rs_asio#224]
I'm having trouble getting a Hotone Jogg audio interface to work.

From looking at the log, it seem like it is failing because the interface only has one ASIO input channel? Does that make sense, and if so, is there any work around?

Thanks for any help!

RS_ASIO.ini:

```
[Config]
EnableWasapi=0
EnableAsio=1

[Asio]
; available buffer size modes:
;    driver - respect buffer size setting set in the driver
;    host   - use a buffer size as close as possible as that requested by the host application
;    custom - use the buffer size specified in CustomBufferSize field
BufferSizeMode=driver
CustomBufferSize=64

[Asio.Output]
Driver=HOTONE AUDIO USB Audio Device
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.0]
Driver=HOTONE AUDIO USB Audio Device
Channel=0
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.1]
Driver=
Channel=1
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100
```

RS_ASOI-log.txt:

```
0.000 [INFO]   - Wrapper DLL loaded (v0.4.0)
0.000 [INFO]  PatchOriginalCode
0.000 [INFO]  Patching CoCreateInstance
0.186 [INFO]  Patch_CallAbsoluteAddress - num locations: 5
0.186 [INFO]  Patching call at 01C5AA85
0.186 [INFO]  Patching call at 01F03B6E
0.186 [INFO]  Patching call at 01F5C681
0.186 [INFO]  Patching call at 01FAEF75
0.186 [INFO]  Patching call at 01FAF40D
0.186 [INFO]  Patching PortAudio MarshalStreamComPointers
0.362 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.362 [INFO]  Patching call at 01F03059
0.362 [INFO]  Patching PortAudio UnmarshalStreamComPointers
0.537 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.537 [INFO]  Patching call at 01F0310E
0.539 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
0.539 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
0.539 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
0.539 [INFO]  AsioHelpers::FindDrivers
0.540 [INFO]    ASIO4ALL v2
0.540 [INFO]    GT-1
0.540 [INFO]    HOTONE AUDIO USB Audio Device
0.540 [INFO]    Realtek ASIO
0.540 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: HOTONE AUDIO USB Audio Device
0.540 [INFO]  Creating AsioSharedHost - dll: c:\program files\hotone audio\usb audio device driver\w10_x64\hotoneusbaudioasio.dll
0.547 [INFO]    ASIO input channels info:
0.547 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: IN 1
0.547 [INFO]    ASIO output channels info:
0.547 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: OUT 1
0.547 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: OUT 2
0.547 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.547 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: HOTONE AUDIO USB Audio Device
0.547 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.547 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 1 render devices, 1 capture devices
0.547 [INFO]    hr: 0
0.547 [INFO]    *ppEndpoint: 17B9DF78
0.547 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
0.547 [INFO]    hr: 0
0.547 [INFO]    *ppEndpoint: 17B9DBE8
0.547 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
0.547 [INFO]    hr: 0
0.547 [INFO]    *ppDevices: 0AE2F1B8
0.547 [INFO]  {ASIO Out} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.547 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetCount
0.547 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetAt - iProp: 0
0.547 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetAt - iProp: 1
0.547 [INFO]  {ASIO Out} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.547 [INFO]  {ASIO Out} DebugWrapperEndpoint::GetDataFlow
0.547 [INFO]    *pDataFlow: eRender
0.547 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.547 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetCount
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 0
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 1
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 2
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 3
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.547 [INFO]  {ASIO IN 0} DebugWrapperEndpoint::GetDataFlow
0.547 [INFO]    *pDataFlow: eCapture
0.547 [INFO]  {ASIO IN 0} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.547 [INFO]  {ASIO IN 0} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.554 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 17
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.554 [ERROR]    rejecting IEEE Float as it's incompatible with current ASIO sample type ASIOSTInt32LSB
0.554 [ERROR]    failed to initialize asio driver with requested format
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.554 [ERROR]    rejecting IEEE Float as it's incompatible with current ASIO sample type ASIOSTInt32LSB
0.554 [ERROR]    failed to initialize asio driver with requested format
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::Initialize - ShareMode: Exclusive Flags: 40000 bufferDuration: 3ms periodicity: 3ms
0.554 [INFO]  {ASIO Out} RSAsioAudioClient::Initialize - host requested buffer duration: 3ms (144 frames)
0.554 [INFO]  {ASIO Out} WAVEFORMATEX
  wFormatTag: fffe
  nChannels: 2
  nSamplesPerSec: 48000
  nAvgBytesPerSec: 384000
  nBlockAlign: 8
  wBitsPerSample: 32
  cbSize: 22
  ext.SubFormat: KSDATAFORMAT_SUBTYPE_PCM
  ext.Samples: 24
  ext.dwChannelMask: 3
0.554 [INFO]  {ASIO Out} RSAsioAudioClient::Initialize - actual buffer duration: 0ms (8 frames)
0.554 [INFO]  AsioSharedHost::Setup - startCount: 0
0.554 [ERROR]    unsupported number of channels: 2
0.554 [ERROR]    wave format not supported on inputs.
0.554 [INFO]    hr: E_FAIL
0.554 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 17
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 17
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Shared
0.554 [ERROR]    shared mode is not supported
0.554 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.554 [INFO]  Destroying AsioSharedHost - dll: c:\program files\hotone audio\usb audio device driver\w10_x64\hotoneusbaudioasio.dll
2.024 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
2.024 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
2.024 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
2.024 [INFO]  AsioHelpers::FindDrivers
2.025 [INFO]    ASIO4ALL v2
2.025 [INFO]    GT-1
2.025 [INFO]    HOTONE AUDIO USB Audio Device
2.025 [INFO]    Realtek ASIO
2.025 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: HOTONE AUDIO USB Audio Device
2.025 [INFO]  Creating AsioSharedHost - dll: c:\program files\hotone audio\usb audio device driver\w10_x64\hotoneusbaudioasio.dll
2.033 [INFO]    ASIO input channels info:
2.033 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: IN 1
2.033 [INFO]    ASIO output channels info:
2.033 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: OUT 1
2.033 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: OUT 2
2.033 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
2.033 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: HOTONE AUDIO USB Audio Device
2.033 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
2.033 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 1 render devices, 1 capture devices
2.033 [INFO]    hr: 0
2.033 [INFO]    *ppEndpoint: 17B9E518
2.033 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
2.033 [INFO]    hr: 0
2.033 [INFO]    *ppEndpoint: 17B9E158
2.033 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
2.033 [INFO]    hr: 0
2.033 [INFO]    *ppDevices: 0AE2F438
2.033 [INFO]  {ASIO Out} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
2.033 [INFO]  {ASIO Out} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
2.033 [INFO]  {ASIO Out} DebugWrapperEndpoint::GetDataFlow
2.033 [INFO]    *pDataFlow: eRender
2.033 [INFO]  {ASIO IN 0} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
2.033 [INFO]  {ASIO IN 0} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
2.033 [INFO]  {ASIO IN 0} DebugWrapperEndpoint::GetDataFlow
2.033 [INFO]    *pDataFlow: eCapture
2.039 [INFO]  DebugDeviceEnum::RegisterEndpointNotificationCallback
3.260 [INFO]  DebugDeviceEnum::UnregisterEndpointNotificationCallback
3.261 [INFO]  Destroying AsioSharedHost - dll: c:\program files\hotone audio\usb audio device driver\w10_x64\hotoneusbaudioasio.dll
3.371 [INFO]   - Wrapper DLL unloaded
```

**Comment by mdias** [comment]
Please try again and repost the log with the latest version of RS_ASIO, there has been some fixes since 0.4.0.

**Comment by mikeoliphant** [comment]
My bad - I should have realized that I wasn't running the latest version. It works fine after updating. Thanks for the quick response!

I've added the Hotone Jogg to the working device list:

https://github.com/mdias/rs_asio/pull/226
